### PR TITLE
Intro video improvements

### PR DESCRIPTION
**Pull request by crudelios**
* Play the logo, the intro video and the credits (in that order) when clicked on the main menu. Click each one to move to the next
* Play the intro music (not ROME1) after intro videos are played

### CODE REVIEW

**Comment by bvschaik** [comment]
Hah, you're just ahead of me, I had a similar implementation in mind for tonight :wink:

File: src/window/intro_video.c [hunk 1]
```diff
@@ -4,22 +4,37 @@
...
+#define NUM_INTRO_VIDEOS 2
 
 static struct {
     int width;
     int height;
+    int current_video;
 } data;
 
-static int init(void)
+static const char *intro_videos[NUM_INTRO_VIDEOS] = { "smk/logo.smk", "smk/intro.smk" };
```
Comment (bvschaik) [inline src/window/intro_video.c]: Also add `smk/credits.smk` (and increase NUM_VIDEOS to 3)

File: src/window/intro_video.c [hunk 2]
```diff
@@ -4,22 +4,37 @@
...
+
+static int start_next_video(void)
 {
-    if (video_start("smk/intro.smk")) {
-        video_size(&data.width, &data.height);
-        video_init();
-        return 1;
+    while(data.current_video < NUM_INTRO_VIDEOS) {
```
Comment (bvschaik) [inline src/window/intro_video.c]: Nitpick: space between `while` and `(`

File: src/window/intro_video.c [hunk 3]
```diff
@@ -33,8 +48,12 @@ static void draw_foreground(void)
 static void handle_mouse(const mouse *m)
 {
     if (m->left.went_up || m->right.went_up || video_is_finished()) {
+        sound_music_stop();
```
Comment (bvschaik) [inline src/window/intro_video.c]: There should be no need to stop music here, since the video player manages its own video.

Comment (crudelios) [reply src/window/intro_video.c]: It is, otherwise you can hear ROME1 start playing between the videos :/

File: src/window/intro_video.c (hunk 1 above)
Comment (crudelios) [reply src/window/intro_video.c]: Right now the videos are playing in succession. If I do that, credits will play right after the intro videos. Are you sure about that?

Comment (bvschaik) [reply src/window/intro_video.c]: Yes. That's what C3 also does.

**Pull request merged**